californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java

```diff
@@ -122,8 +126,11 @@
 
 	private String scheme;
 
-	/** the authenticated (remote) sender's identity */
-	private Principal senderIdentity;
+	/** The destination address of this message. */
+	private InetAddress destination;
+
+	/** The destination port of this message. */
+	private int destinationPort;
```

Comment:
"setDestination(InetAddress)" and "setDestinationPort(int)" have been part of the API.
 I would like to support them for a while. Therefore I deprecated the setter.

Reply:
I think about internal attribute, not external API.
But rethinking about that, setting just inetAddress or just port would be difficult without new internal attribute.

Reply:
Yes. 
On the one hand, I don't want to spent too much time in something I want to remove in the future.
And on the other hand, I want to make the "EndpointContext" to be unmodifiable (because it's hand over by threads).